### ISSUE TITLE
Hide channel messages number in sidebar & show only mentions

### ISSUE THREAD

**Issue by Razva** [thelounge/thelounge#3070]
### Feature Description
Right now TheLounge is showing, in the sidebar, the total number of unread messages. This is very annoying if one is interested about messages addressed exclusively to him, especially in channels with big traffic and lots of messages.

A setting allows displaying the number of messages exclusively addressed to `@my_own_nickname` is ideal in such a scenario, avoiding the chore of just clicking through each channel just to "clear the numbers".

**Comment by xPaw** [comment]
You can easily achieve this with CSS, something like this should work:
```css
#sidebar .badge:not(.highlight) {
  display: none;
}```

**Comment by Razva** [comment]
> You can easily achieve this with CSS, something like this should work:
> 
> ```css
> #sidebar .badge:not(.highlight) {
>   display: none;
> }
> ```

Thanks for the hint, but it seems that it's not working as expected. Example:
1. @user1 is mentioned - no number is displayed in sidebar
2. @user1 is mentioned again - no number is displayed in sidebar
3. @myself is mentioned - "3" is posted in the sidebar, when just "1" should be posted

**Comment by xPaw** [comment]
Yes that will happen because it keeps the count of unread numbers. You could do some CSS trickery to hide the number and display something else.

**Comment by Razva** [comment]
> Yes that will happen because it keeps the count of unread numbers. You could do some CSS trickery to hide the number and display something else.

I fully agree with you. Still, shouldn't this be a core feature?

**Comment by xPaw** [comment]
I don't see why it needs to be one.

**Comment by Razva** [comment]
> I don't see why it needs to be one.

Because it's very annoying to have 4k "unread messages" each morning, for 20 channels, and when online constantly jump between channels to clear "the number" just for "it" to grow again after a couple of hours. I agree with the fact that some users might enjoy it, but some - as myself - don't, so a "show number only for mentions" option would be awesome.

**Comment by dgw** [comment]
I was all set to go make a screenshot of Textual's channel properties dialog, showing how it lets you mark a channel to show only mentions… but it doesn't support that. My memory was incorrect (or I just have an old version, maybe).

Agreed that this would be a useful feature to have, for busy channels. But I have no strong feeling one way or the other on whether this should be part of The Lounge core, or relegated to a plugin (assuming the plugin API will support doing such things, that is).

As it stands, implementing this as a global setting would probably not cover all use cases. Until The Lounge has support for per-channel settings (correct me if I'm wrong in thinking that it doesn't already), this probably isn't really doable even in core. Until the plugin API actually exists, it's not doable as a plugin.

**Comment by gregsterb** [comment]
Does the lounge have plugins?

**Comment by McInkay** [comment]
> Does the lounge have plugins?

Yeah, but the API is reasonably bare-bones right now.

**Comment by Jay2k1** [comment]
Late follow-up, but since this came up on IRC recently, a custom CSS hack that's a bit better than just hiding the unread counter badge. This also hides the unread counter badge, and for the highlight badge, it hides the (misleading) number on it so you just have sort of an indicator icon:

![image](https://user-images.githubusercontent.com/2070503/102111888-3d72f700-3e37-11eb-82c1-740266f8e8bb.png)


```css
#sidebar .badge:not(.highlight) { 
    display: none;
}
#sidebar .badge.highlight {
    color: white; /* or whatever background color your highlight badge has, depending on your theme */
    border-radius: 20px; /* optional, makes the badge circle shaped */
}
```
